### PR TITLE
Add initial command guidelines and make all commands follow them.

### PR DESCRIPTION
**Pull request by BillyONeal**
Remaining work (to be done in separate PRs):

Missing help: 
acquire-project
acquire
activate
generate-msbuild-props
use

Missing autocomplete consistency.

Missing vcpkgcmdarguments print_command_list_usage consistency.

### CODE REVIEW

File: src/vcpkg-test/commands.extract.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #include <vcpkg-test/util.h>
 
 using namespace vcpkg;
-using namespace vcpkg::Commands;
+using namespace vcpkg;
```
Comment (BillyONeal) [inline src/vcpkg-test/commands.extract.cpp:15]:
```suggestion

```

File: src/vcpkg/commands.check-support.cpp [hunk 2]
```diff
@@ -14,102 +14,105 @@
...
+                .append(msgUnsupportedPortDependency, msg::value = full_port_name(p))
+                .append(msgPortSupportsField, msg::supports_expression = reason.supports_expr);
+        }
+        msg::println(message);
     }
+} // unnamed namespace
+
+namespace vcpkg
+{
+    static constexpr StringLiteral OPTION_JSON{"x-json"};
```
Comment (BillyONeal) [inline src/vcpkg/commands.check-support.cpp:97]: Move up

**Comment by autoantwort** [comment]
Should we add a field `description` to the `CommandMetadata`? Then you can more or less autogenerate print_command_list_usage and the "help" text could print what the command does (I mean the text if you execute for example `vcpkg install --help`).

**Comment by BillyONeal** [comment]
> Should we add a field `description` to the `CommandMetadata`? Then you can more or less autogenerate print_command_list_usage and the "help" text could print what the command does (I mean the text if you execute for example `vcpkg install --help`).

Yes, this is in the cards. I was just already editing most of the things here and was already about causing peoples eyes to bleed trying to review it.

**Review by JavierMatosD — APPROVED** [review]

**Pull request merged**